broadleaf-thymeleaf2-presentation/src/main/java/org/broadleafcommerce/presentation/thymeleaf2/resolver/DelegatingResourceResolver.java

```diff
@@ -0,0 +1,52 @@
+/*
+ * #%L
+ * broadleaf-theme
+ * %%
+ * Copyright (C) 2009 - 2016 Broadleaf Commerce
+ * %%
+ * Licensed under the Broadleaf Fair Use License Agreement, Version 1.0
+ * (the "Fair Use License" located  at http://license.broadleafcommerce.org/fair_use_license-1.0.txt)
+ * unless the restrictions on use therein are violated and require payment to Broadleaf in which case
+ * the Broadleaf End User License Agreement (EULA), Version 1.1
+ * (the "Commercial License" located at http://license.broadleafcommerce.org/commercial_license-1.1.txt)
+ * shall apply.
+ * 
+ * Alternatively, the Commercial License may be replaced with a mutually agreed upon license (the "Custom License")
+ * between you and Broadleaf Commerce. You may not use this file except in compliance with the applicable license.
+ * #L%
+ */
+package org.broadleafcommerce.presentation.thymeleaf2.resolver;
+
+import org.broadleafcommerce.presentation.resolver.BroadleafTemplateResolver;
+import org.springframework.stereotype.Service;
+import org.thymeleaf.TemplateProcessingParameters;
+import org.thymeleaf.resourceresolver.IResourceResolver;
+
+import java.io.InputStream;
+
+
+/**
+ * An implementation of {@link IResourceResolver} that provides an extension point for retrieving
+ * templates from the database.
+ * 
+ * @author Jon Fleschler (jfleschler)
+ */
+@Service("blDelegatingResourceResolver")
+public class DelegatingResourceResolver implements IResourceResolver {
+    protected BroadleafTemplateResolver templateResolver;
+
+    public DelegatingResourceResolver(BroadleafTemplateResolver templateResolver) {
+        super();
+        this.templateResolver = templateResolver;
+    }
+
+    @Override
+    public String getName() {
+        return "BL_CUSTOM";
+    }
+
+    @Override
+    public InputStream getResourceAsStream(TemplateProcessingParameters params, String resourceName) {
```

Comment:
I only included the `templateName` and `resourceName` as parameters because from what I could tell, those were the only pieces we ever used for custom template resolving.  That and they were the two parameters that overlapped the most between TL2 and TL3.  The only other piece that might be useful is the locale .. but I couldn’t find it on the TL3 side.